src/object/obj_rpc.c

```diff
@@ -878,22 +878,11 @@ crt_proc_struct_daos_cpd_bulk(crt_proc_t proc, crt_proc_op_t proc_op,
 	if (unlikely(rc))
 		return rc;
 
-	if (DECODING(proc_op)) {
-		D_ALLOC_PTR(dcb->dcb_bulk);
-		if (dcb->dcb_bulk == NULL)
-			return -DER_NOMEM;
-	}
-
-	rc = crt_proc_crt_bulk_t(proc, proc_op, dcb->dcb_bulk);
-	if (unlikely(rc))
-		return rc;
-
-	if (FREEING(proc_op))
-		D_FREE(dcb->dcb_bulk);
+	rc = crt_proc_crt_bulk_t(proc, proc_op, &dcb->dcb_bulk);
```

Comment:
I think this change will cause compatibility issue, no?

Reply:
No, the on-wire layout will be the same, transferring the bulk handle for both cases. But it simplifies the local structure and related logic.

Reply:
No, the on-wire layout will be the same, transferring the bulk handle for both cases. But it simplifies the local structure and related logic.

Reply:
No, the on-wire layout will be the same, transferring the bulk handle for both cases. But it simplifies the local structure and related logic.